src/inmanta/util/dict_path.py

```diff
@@ -667,6 +667,15 @@ def __add__(self, other: object) -> "DictPath":
             return NotImplemented
         return ComposedPath(path=list(self.get_path_sections()) + list(other.get_path_sections()))
 
+    def __contains__(self, other: "DictPath") -> bool:
+        other_key_path_sections = [e.get_key() for e in other.get_path_sections()]
+        self_key_path_sections = [e.get_key() for e in self.get_path_sections()]
+        min_len = min(len(other_key_path_sections), len(self_key_path_sections))
+        for i in range(min_len):
+            if self_key_path_sections[i] != other_key_path_sections[i]:
+                return False
+        return True
+
```

Comment:
I agree with @edvgui's sentiment: I would first like a concrete proposal of what exactly the semantics are, with tests/examples for all possible cases you can think of. One more I just thought of:
```
assert to_path("a.b") in to_path("a.b[a=1]") # I think, not very confident
```

For the ones @edvgui posted, I think `"b.c" in "a.b.c"` is the most debatable: does `in` mean that one selector goes strictly deeper than the other (i.e. it starts the same but is longer)? Or does it mean that the left selector is some part of the right one (as @edvgui's example)?

Also, are wildcards ever allowed? In the left operand or the right one? How does `NullPath` come into it?

Reply:
I also think this will not work correctly on e.g.
```
assert to_path("a.b.c") in to_path("a.b")
```

as the algorithm is symmetric, but the expected behavior is not.